data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndRawSpanTest.java

```diff
@@ -135,6 +137,66 @@ public void testPipelineEndToEnd() throws InterruptedException {
         );
     }
 
+    @Test
+    public void testPipelineCompatibilityEndToEnd() throws InterruptedException {
+        //Send data to otel trace source
+        final ExportTraceServiceRequest exportTraceServiceRequestTrace1BatchWithRoot = getExportTraceServiceRequest(
+                getResourceSpansBatch(TEST_SPAN_SET_1_WITH_ROOT_SPAN)
+        );
+        final ExportTraceServiceRequest exportTraceServiceRequestTrace1BatchNoRoot = getExportTraceServiceRequest(
+                getResourceSpansBatch(TEST_SPAN_SET_1_WITHOUT_ROOT_SPAN)
+        );
+        final ExportTraceServiceRequest exportTraceServiceRequestTrace2BatchWithRoot = getExportTraceServiceRequest(
+                getResourceSpansBatch(TEST_SPAN_SET_2_WITH_ROOT_SPAN)
+        );
+        final ExportTraceServiceRequest exportTraceServiceRequestTrace2BatchNoRoot = getExportTraceServiceRequest(
+                getResourceSpansBatch(TEST_SPAN_SET_2_WITHOUT_ROOT_SPAN)
+        );
+
+        sendExportTraceServiceRequestToSource(DATA_PREPPER_MAIN_PORT, exportTraceServiceRequestTrace1BatchWithRoot);
+        sendExportTraceServiceRequestToSource(DATA_PREPPER_MAIN_PORT, exportTraceServiceRequestTrace2BatchNoRoot);
+        sendExportTraceServiceRequestToSource(DATA_PREPPER_MAIN_PORT, exportTraceServiceRequestTrace1BatchNoRoot);
```

Comment:
nit - I noticed the 3rd request is sent to `DATA_PREPPER_PORT_2` in the original test, any reason for it to differ here?

Reply:
Yes, it is due to OTelTraceGroupPrepper missing in the latest pull image. Thus the whole traceGroup 1 is directed towards the current bleeding edge.